systemPrompts/developer.py

```diff
@@ -0,0 +1,100 @@
+import re
+import inspect
+
+def intercept(fstring):
+    frame = inspect.currentframe().f_back
+    code = frame.f_code
+    filename = code.co_filename
+    line_number = code.co_firstlineno
+    locals_dict = frame.f_locals
+
+    # Extract expressions within the f-string
+    regex = r'\{([^{}]+)\}'
+    expressions = re.findall(regex, fstring)
+
+    # Process the expressions
+    components = []
+    for expr in expressions:
+        # Unescape braces
+        expr = expr.replace('{{', '{').replace('}}', '}')
+
+        # Evaluate the expression
+        result = eval(expr, globals(), locals_dict)
+        components.append(result)
+
+    # Log or perform any desired actions with the components
+    print("Intercepted components:", components)
+    print("File:", filename)
+    print("Line number:", line_number)
+
+    # Pass through the f-string as-is
+    return fstring
+
+
+def call(fstring):
+    print("Original f-string:", fstring)
+
+
+
+
+def planPrompt1():
+  return """You are an AI developer who is trying to write a program that will generate code for the user based on their intent.
+
+  When given their intent, create a complete, exhaustive list of filepaths that the user would write to make the program.
+
+  only list the filepaths you would write, and return them as a python list of strings.
+  do not add any other explanation, only return a python list of strings.
+  """
+  
+def planPrompt2(sourcePrompt, filepaths_string):
+  
+  return """You are an AI developer who is trying to write a program that will generate code for the user based on their intent.
```

Comment:
Hi, I think you forgot to put `f` before the quotes (like this `f"""`). But be careful, if you decide to fix it - you don't want the `{prompt}` to appear twice in the request to OpenAI (because you also add `prompt` as a user message right after this system message).

Generally speaking, I would highly recommend using a tool like PromptLayer to keep track of what is actually sent to OpenAI. A hint on how it can be added to this project can be found here: https://github.com/smol-ai/developer/pull/74/files

Reply:
thank you! i have made my own no api needed version :) https://github.com/smol-ai/logger

Reply:
Hey! A tool that logs LLM prompts and responses locally without sending the logs to the cloud sounds like a great idea! Can your tool be used from Python, or is it purely for Javascript projects?